### PR TITLE
[TF-TRT] Throwing error if compiling with NVIDIA TensorRT < 7

### PR DESCRIPTION
**Pull request by DEKHTIARJonathan**
This PR blocks TF compilation with TRT_VERSION < 7, effectively removing support for TRT 5 and 6.

### CODE REVIEW

**Comment by DEKHTIARJonathan** [comment]
I had to re-run a few times to fix some PyLint issues

**Comment by DEKHTIARJonathan** [comment]
@bixia1 any comment on the PR ? Or we are good to merge as soon as we get approval from Pankaj & Sanjoy ?

**Comment by bixia1** [comment]
I put some detail on our weekly meeting doc. It was decided that we don't need to issue such a warning as TF nightly build have been switch to TRT 7 a while back; and we will do the code change for removing the support at the end of June.

**Comment by DEKHTIARJonathan** [comment]
@bixia1 modifications executed as requested.

Feel free to rapidly test: https://www.onlinegdb.com/online_c++_compiler

```cpp
/******************************************************************************

                              Online C++ Compiler.
               Code, Compile, Run and Debug C++ program online.
Write your code in this editor and press "Run" button to compile and execute it.

*******************************************************************************/

#include <iostream>

#define NV_TENSORRT_MAJOR 6
#define NV_TENSORRT_MINOR 0
#define NV_TENSORRT_PATCH 0
#define NV_TENSORRT_BUILD 0


#define IS_TRT_VERSION_GE(major, minor, patch, build)           \
  ((NV_TENSORRT_MAJOR > major) ||                               \
   (NV_TENSORRT_MAJOR == major && NV_TENSORRT_MINOR > minor) || \
   (NV_TENSORRT_MAJOR == major && NV_TENSORRT_MINOR == minor && \
    NV_TENSORRT_PATCH > patch) ||                               \
   (NV_TENSORRT_MAJOR == major && NV_TENSORRT_MINOR == minor && \
    NV_TENSORRT_PATCH == patch && NV_TENSORRT_BUILD >= build))

#if ! IS_TRT_VERSION_GE(7, 0, 0, 0)
    
    #error From TensorFlow 2.6, we only support NVIDIA TensorRT version 7 and older.
    #error Please update your environment and relaunch the compilation.

#endif

using namespace std;

int main()
{
    cout<<"Hello World";

    return 0;
}
```

File: tensorflow/compiler/tf2tensorrt/convert/trt_optimization_pass.h
Comment (bixia1) [inline tensorflow/compiler/tf2tensorrt/convert/trt_optimization_pass.h:29]: Do you mean s/older/newer/?

Also, can you fix the description of the PR? Right now you have
Adding deprecation warning for TRT < 7

Since this is a build error, shall we s/warning/error/ ?

Comment (DEKHTIARJonathan) [reply tensorflow/compiler/tf2tensorrt/convert/trt_optimization_pass.h]: Both comments fixed

**Comment by DEKHTIARJonathan** [comment]
@bixia1 comments addressed

File: tensorflow/compiler/tf2tensorrt/convert/trt_optimization_pass.h
Comment (bixia1) [inline tensorflow/compiler/tf2tensorrt/convert/trt_optimization_pass.h:30]: Sorry to pick on this again: the PR subject says "warning" here you use error.
If we want a warning, let's change #error here to #warning

Comment (DEKHTIARJonathan) [reply tensorflow/compiler/tf2tensorrt/convert/trt_optimization_pass.h:30]: Oh I see my bad, I didn't see notice this one. I fixed the title of the PR.

**Review by bixia1 — APPROVED** [review]

**Pull request merged**
